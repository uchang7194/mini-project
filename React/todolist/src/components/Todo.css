.todo {
  padding: 6px 10px;
  list-style: none;
  box-shadow: 2px 2px 2px 2px rgba(100, 100, 100, 0.65);
  border-radius: 10px;
}
.todo:hover {
  background: #ddd;
}
.todo::after {
  content: '';
  display: block;
  clear: both;
}
.todo p {
  margin: 0;
}
.todo .show-todo {
  width: 80%;
  float: left;
  margin: 0;
}
.todo .show-todo > input[type="text"] {
  width: 100%;
}
.todo button[type="button"] {
  padding: 0;
  border: 0 none;
  background: transparent;
  cursor: pointer;
  margin-right: 8px;
}
.todo button[type="button"]:last-child {
  margin-right: 0;
}
.todo .utils {
  float: right;
  width: 20%;
  text-align: right;
}