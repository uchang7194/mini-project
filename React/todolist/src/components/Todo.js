import React, { Component } from 'react'
import PropTypes from 'prop-types';

import './Todo.css';


export default class Todo extends Component {

  constructor(props) {
    super(props);

    this.state = {
      value: ''
    }
  }

  _handleOnClickDelete = () => {
    
    let confirm = window.confirm('삭제 하시겠습니까?');

    if( confirm ) {
      this.props.removeTodo(this.props.idx);
    }
  }

  _handleOnClickModified = () => {
    this.props.activeModifiedTodo(this.props.idx);
  }

  _handleOnChange = (e) => {
    const value = e.target.value;

    this.setState({
      value
    });
  }

  _handleOnKeyDown = (e) => {
    const value = e.target.value;
    const keyCode = e.keyCode;

    if( keyCode === 13 ) {
      e.preventDefault();
      this.props.modifiedTodo(this.props.idx, value);
    }
  }

  _renderTodoText = () => {
    const todo = this.props.todo;
    if( !this.props.modified_flag ) {
      return (
        <p>{todo}</p>
      );
    } else {
      return (
        <input 
          type="text" 
          value={this.state.value} 
          onChange={ (e) => { this._handleOnChange(e) }}
          onKeyDown={ (e) => { this._handleOnKeyDown(e) }}
          />
      )
    }
  }
  componentWillReceiveProps = (newProps) => {
    this.setState({
      value: newProps.todo
    });
  }
  render() {
    return (
      <li className="todo">
        <div className="show-todo">
          { this._renderTodoText() }
        </div>
        <div className="utils">
          <button type="button" onClick={() => { this._handleOnClickDelete() }}>삭제</button>
          <button type="button" onClick={() => { this._handleOnClickModified() }}>수정</button>
        </div>
      </li>
    );
  }
}

Todo.PropTypes = {
  todo: PropTypes.string.isRequired
}
