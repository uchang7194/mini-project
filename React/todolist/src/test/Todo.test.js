import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Todo from '../components/Todo';

it('toJSON으로 파일이 어떻게 스냅샷이 되는가?', () => {
  const component = renderer.create(<Todo />);
  let tree = component.toJSON();

  console.log(tree);
  expect(tree).toMatchSnapshot();

  tree.props.onChange();

  console.log(tree);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  console.log(tree);


});