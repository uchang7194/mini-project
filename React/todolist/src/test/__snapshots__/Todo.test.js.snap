// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toJSON으로 파일이 어떻게 스냅샷이 되는가? 1`] = `
<li
  className="todo"
>
  <div
    className="show-todo"
  >
    <p />
  </div>
  <div
    className="utils"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      삭제
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      수정
    </button>
  </div>
</li>
`;
